### PR TITLE
Moved nupic extensions from nupic to nupic.core and updated setup.py

### LINKED ISSUE #2312: Move PyRegion and py_support extensions from nupic to nupic.core
This is part of a larger plan to move all the C++ from nupic to nupic.core. https://github.com/numenta/nupic/issues/2303

### PR DESCRIPTION
**Pull request by akhilaananthram**
Moved PyRegion and the py_support helper files from nupic to nupic.core. Also updated setup.py since we no longer needed a dynamic library for PyRegion.  Updated the paths to py_support in setup.py and the nupic/bindings.

Changes done to nupic.core: https://github.com/numenta/nupic.core/pull/492

Fixes #2312

@scottpurdy

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -404,27 +397,16 @@ def getExtensionModules(nupicCoreReleaseDir, platform, bitness, cmdOptions=None)
...
       getLibPrefix(platform) + "nupic_core" + getStaticLibExtension(platform)]
 
+  # Find py_support .cpp files
+  nupic_core = os.path.dirname(os.path.dirname(os.path.dirname(nupicCoreReleaseDir)))
   pythonSupportSources = [
-    "extensions/py_support/NumpyVector.cpp",
-    "extensions/py_support/PyArray.cpp",
-    "extensions/py_support/PyHelpers.cpp",
-    "extensions/py_support/PythonStream.cpp"]
+    nupicCoreReleaseDir + 'include/nupic/py_support/NumpyVector.cpp',
```
Comment (scottpurdy) [inline setup.py]: keep double quotes

**Comment by scottpurdy** [comment]
Please squash commit history into logical units with clear commit messages.

**Comment by scottpurdy** [comment]
Done with review

**Comment by scottpurdy** [comment]
Now there are 42 commits in this PR...

**Comment by akhilaananthram** [comment]
@scottpurdy: please review

**Comment by akhilaananthram** [comment]
@rhyolight: what should I do about the appveyor problem for nupic? It looks like it doesn't even attempt to build it

**Comment by rhyolight** [comment]
@akhilaananthram I have it turned off. I'll turn it back on now!

**Comment by rhyolight** [comment]
@akhilaananthram The appveyor build is running and failing: https://ci.appveyor.com/project/numenta-ci/nupic/build/1243. It needs to be set up properly. Ignore it for now, but https://github.com/numenta/nupic/issues/1888 still needs to be fixed eventually. I think we are still in the [build phase](http://www.appveyor.com/docs/build-phase) of setting up Appveyor.

**Comment by akhilaananthram** [comment]
Ok, thanks! @scottpurdy: please review.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**